plugins/templates/removeTag.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+const boom = require('boom');
+const schema = require('screwdriver-data-schema');
+
+/* Currently, only build scope is allowed to tag template due to security reasons.
+ * The same pipeline that publishes the template has the permission to tag it.
+ */
+module.exports = () => ({
+    method: 'DELETE',
+    path: '/templates/tags',
+    config: {
+        description: 'Delete a template tag',
+        notes: 'Delete a specific template',
+        tags: ['api', 'templates'],
+        auth: {
+            strategies: ['token', 'session'],
+            scope: ['build']
+        },
+        plugins: {
+            'hapi-swagger': {
+                security: [{ token: [] }]
+            }
+        },
+        handler: (request, reply) => {
+            const pipelineFactory = request.server.app.pipelineFactory;
+            const templateFactory = request.server.app.templateFactory;
+            const templateTagFactory = request.server.app.templateTagFactory;
+            const pipelineId = request.auth.credentials.pipelineId;
+            const config = request.payload;
+
+            return templateTagFactory.get({
+                name: config.name,
+                tag: config.tag
+            })
+            .then((templateTag) => {
+                if (!templateTag) {
+                    throw boom.notFound('Template tag does not exist');
+                }
+
+                return Promise.all([
+                    pipelineFactory.get(pipelineId),
+                    templateFactory.get({
+                        name: config.name,
+                        version: templateTag.version
+                    })
+                ])
+                .then(([pipeline, template]) => {
+                    // Check for permission
```

Comment:
I don't think it's necessary because of our logic. `templatetag` can't be created if the `template` itself doesn't exist. There is already [a check if `templateTag` exists]( https://github.com/screwdriver-cd/screwdriver/blob/02434e10cb28eb81214a33821b937d7a70f33c22/plugins/templates/removeTag.js#L37).  If `templateTag` exists, then `template` should exist.